tests/nested/core/core-revert/task.yaml

```diff
@@ -28,37 +28,37 @@ execute: |
     #shellcheck source=tests/lib/nested.sh
     . "$TESTSLIB/nested.sh"
 
-    if [ "$BUILD_SNAPD_FROM_CURRENT" = "true" ] || [ "$CORE_CHANNEL" == "$CORE_REFRESH_CHANNEL" ]; then
+    if [ "$NESTED_BUILD_SNAPD_FROM_CURRENT" = "true" ] || [ "$NESTED_CORE_CHANNEL" == "$NESTED_CORE_REFRESH_CHANNEL" ]; then
         echo "This test need to start from beta/candidate/stable channel"
         exit 0
     fi
 
-    execute_remote "snap info core" | MATCH "tracking: +latest/${CORE_CHANNEL}"
-    execute_remote "sudo snap refresh --${CORE_REFRESH_CHANNEL} core" || true
+    nested_execute "snap info core" | MATCH "tracking: +latest/${NESTED_CORE_CHANNEL}"
+    nested_execute "sudo snap refresh --${NESTED_CORE_REFRESH_CHANNEL} core" || true
 
-    if ! wait_for_ssh; then
+    if ! nested_wait_for_ssh; then
         echo "ssh not stablished, exiting..."
         exit 1
     fi
 
-    while ! execute_remote "snap changes" | MATCH "Done.*Refresh \"core\" snap from \"${CORE_REFRESH_CHANNEL}\" channel"; do
+    while ! nested_execute "snap changes" | MATCH "Done.*Refresh \"core\" snap from \"${NESTED_CORE_REFRESH_CHANNEL}\" channel"; do
```

Comment:
Same problem here.

Reply:
I was fixed before, I see that is being used the variable NESTED_CORE_REFRESH_CHANNEL there

Reply:
I'm talking about the use of a function under while. Negation is also a factor.

Reply:
Yes, It is in the list for the following PR. thanks